packetbeat/pb/event.go

```diff
@@ -446,6 +455,11 @@ func marshalStruct(m mapstr.M, key string, val reflect.Value) error {
 			if err := marshalStruct(m, key, fieldValue); err != nil {
 				return err
 			}
+			// assume a pointer is a struct or other object we can marshal
+		} else if structField.Type.Kind() == reflect.Ptr || structField.Type.Kind() == reflect.Struct {
```

Comment:
We probably should not assume that any pointer will be a struct pointer. I suggest adding a check like

`(structField.Type.Kind() == reflect.Ptr && fieldValue.Elem().Kind() == reflect.Struct) || ...`

Reply:
Wouldn't that effectively be the same check as the one you suggested above?